src/system-program.js

```diff
@@ -2,9 +2,154 @@
 
 import * as BufferLayout from 'buffer-layout';
 
-import {Transaction} from './transaction';
+import {Transaction, TransactionInstruction} from './transaction';
 import {PublicKey} from './publickey';
 import * as Layout from './layout';
+import type {TransactionInstructionCtorFields} from './transaction';
+
+/**
+ * System Instruction class
+ */
+export class SystemInstruction extends TransactionInstruction {
+  /**
+   * Type of SystemInstruction
+   */
+  type: SystemInstructionType;
+
+  constructor(
+    opts?: TransactionInstructionCtorFields,
+    type?: SystemInstructionType,
+  ) {
+    if (
+      opts &&
+      opts.programId &&
+      !opts.programId.equals(SystemProgram.programId)
+    ) {
+      throw new Error('programId incorrect; not a SystemInstruction');
+    }
+    super(opts);
+    if (type) {
+      this.type = type;
+    }
+  }
+
+  static from(instruction: TransactionInstruction): SystemInstruction {
+    if (!instruction.programId.equals(SystemProgram.programId)) {
+      throw new Error('programId incorrect; not SystemProgram');
+    }
+
+    const instructionTypeLayout = BufferLayout.u32('instruction');
+    const typeIndex = instructionTypeLayout.decode(instruction.data);
+    let type;
+    for (const t in SystemInstructionEnum) {
+      if (SystemInstructionEnum[t].index == typeIndex) {
+        type = SystemInstructionEnum[t];
+      }
+    }
+    if (!type) {
+      throw new Error('Instruction type incorrect; not a SystemInstruction');
+    }
+    return new SystemInstruction(
+      {
+        keys: instruction.keys,
+        programId: instruction.programId,
+        data: instruction.data,
+      },
+      type,
+    );
+  }
+
+  /**
+   * The `from` public key of the instruction;
+   * returns null if SystemInstructionType does not support this field
+   */
+  get From(): PublicKey | null {
```

Comment:
Can we rename `From` to maybe fromPubkey to avoid confusion with the the `from` static method.  Other names work too.   Also the accessor methods `From`/`To`/`Amount` should start with a lowercase letter

Reply:
(or rename the `from` static method to something else)